server/src/Log.ts

```diff
@@ -25,6 +25,9 @@ export class Log {
 
     private static lastProgress: number;
 
+    /**
+     * Do not forget to call `startProgress()` before the first call to `progress`
```

Comment:
Seems that this could be simplified, e.g. if the code checks that this is the first call to ```progress``` then one thing happens, and otherwise another thing happens. Not an urgent fix though.